flottform/demo/src/routes/custom-ui/+page.svelte

```diff
@@ -0,0 +1,586 @@
+		const letters = valueToFill.split('');
+		let i = 0;
+		$prefilledForm[refValue] = '';
+		while (i < letters.length) {
+			await waitForMs(delay);
+			$prefilledForm[refValue] += letters[i];
+			i++;
+		}
+		return;
+	}
+
+	function waitForMs(ms: number) {
+		return new Promise((resolve) => setTimeout(resolve, ms));
+	}
+
+	const fillOutForm = async () => {
+		resetValues();
+		isFillingOut = true;
+		await typeValue(randomArrayElement(mockUser.name), 'name');
+		await waitForMs(100);
+		await typeValue(randomArrayElement(mockUser.surname), 'surname');
+		await waitForMs(50);
+		await typeValue(
+			`${$prefilledForm.name}.${$prefilledForm.surname}@example.com`.toLocaleLowerCase(),
+			'email'
+		);
+		await waitForMs(80);
+		await typeValue(randomArrayElement(mockUser.phone), 'phone');
+		await waitForMs(110);
+		await typeValue(randomArrayElement(mockUser.street), 'street');
+		await waitForMs(40);
+		await typeValue(randomArrayElement(mockUser.houseNumber), 'houseNumber');
+		await waitForMs(80);
+		await typeValue(randomArrayElement(mockUser.city), 'city');
+		await waitForMs(50);
+		await typeValue(randomArrayElement(mockUser.postcode), 'postcode');
+		await waitForMs(70);
+		await typeValue(randomArrayElement(mockUser.problemDescription), 'problemDescription', 5);
+		highlighted = true;
+		isFillingOut = false;
+	};
+
+	onMount(async () => {
+		const fileInput = document.querySelector('input[type=file]') as HTMLInputElement;
+
+		const flottformFileInputHost = new FlottformFileInputHost({
+			flottformApi: sdpExchangeServerBase,
+			createClientUrl,
+			inputField: fileInput
+		});
+
+		flottformFileInputHost.on('new', () => {
+			const createChannelLinkDialog = createDialogCard(flottformFileInputHost, defaultStyles);
+			document.body.appendChild(createChannelLinkDialog);
+
+			const createChannelElement = createFlottformButtonParentElement(
+				fileInput,
+				createChannelLinkDialog,
+				flottformFileInputHost,
+				defaultStyles
+			);
+			fileInput?.after(createChannelElement);
+		});
+
+		flottformFileInputHost.on('endpoint-created', ({ link, qrCode }) => {
+			const {
+				createChannelQrCode,
+				createChannelLinkWithOffer,
+				createDialogDescription,
+				createChannelButton,
+				createChannelStatusWrapper
+			} = getElements();
+
+			createChannelButton.removeEventListener('click', flottformFileInputHost.start);
+			createChannelButton.style.background = changeBackgroundOnState(
+				'waiting-for-client',
+				defaultStyles
+			);
+
+			createChannelQrCode.style.display = 'block';
+			createChannelQrCode.setAttribute('src', qrCode);
+
+			createChannelLinkWithOffer.style.display = 'block';
+			createChannelLinkWithOffer.setAttribute('href', link);
+			createChannelLinkWithOffer.innerHTML = link;
+
+			createChannelStatusWrapper.innerHTML = 'Upload a file';
+
+			createDialogDescription.innerHTML = 'Use this QR-Code or Link on your other device.';
```

Comment:
Here is the same: all html related code is already in our `dialog` element, we don't need to create or change it here. In this event we can remove event listener from the button (so it won'r restart the connection) and pass link and qr code to our variables

```suggestion
			createChannelButton.removeEventListener('click', flottformFileInputHost.start);
			webRtcLink = link;
			webRtcQrCode = qrCode;
```